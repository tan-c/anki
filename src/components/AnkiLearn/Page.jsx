import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import moment from 'moment';
import keydown from 'react-keydown';
// import toastr from 'toastr';
// import Highlight from 'react-highlight'; // FIXME: removed as it introduced an older version of react
import { Map, fromJS } from 'immutable';
import {
  Card, Grid, Button, Header, Container, Divider
} from 'semantic-ui-react';

import {
  AnkiActions,
  currentAnkiSelector,
  revisionAnkisTotalSelector
} from 'utility-redux/anki';

export class AnkiPage extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.toggleAnswer = this.toggleAnswer.bind(this);
    this.updateAnki = this.updateAnki.bind(this);
  }

  state = {
    showAnswer: false
  };

  @keydown('space')
  toggleAnswer(event) {
    const { showAnswer } = this.state;
    this.setState({
      showAnswer: !showAnswer
    });
    // if (!showAnswer) {
    //   this.setState({
    //     showAnswer: true,
    //   });
    // } else {
    //   this.updateAnki(null, 2);
    // }
  }

  @keydown('1', '2')
  updateAnki(event, key = undefined) {
    if (this.state.showAnswer) {
      const nth = key === undefined ? event.which - 48 : key; // key value to number
      const { currentAnki } = this.props;

      const revision = currentAnki.get('revision').toJS();

      switch (nth) {
      case 1: // pass to tomorrow
        if (revision.round >= 1) {
          revision.passing += 1;
        }

        if (revision.round >= 5) {
          // Extra penalty for old stuff
          revision.passing += 1;
        }
        revision.nextTime = moment()
          .add(1, 'day')
          .toDate();
        break;
      case 2: // good
        revision.nextTime = moment()
          .add(
            (currentAnki.type === 'anki' ? 1.732 : 2.23) ** revision.round,
            'day'
          )
          .toDate();
        revision.round += 1;
        break;
      default:
        break;
      }

      if (revision.passing > 4 && revision.round >= 1) {
        revision.round -= 1;
        revision.passing -= 4;
      }

      this.setState(prevState => ({
        showAnswer: false
      }));

      const newCurrentAnki = currentAnki.mergeDeep(
        fromJS({
          revision
        })
      );

      this.props.AnkiActions.update(
        newCurrentAnki.set('lastUpdate', new Date())
      );
    }
  }

  render() {
    const { currentAnki, revisionAnkisTotal } = this.props;
    const { showAnswer } = this.state;

    let revisionElapsedDays = 0;
    let createdElapsedDays = 0;
    // update the revision time to be x days from now
    if (currentAnki.has('lastUpdate') !== undefined) {
      const revisionElapsedDaysUnparsed = (new Date().getTime()
        - new Date(currentAnki.get('lastUpdate')).getTime())
        / 60
        / 60
        / 1000
        / 24;
      revisionElapsedDays = parseInt(revisionElapsedDaysUnparsed, 10) + 1;
    }

    const createdElapsedDaysUnparsed = (new Date().getTime()
      - new Date(currentAnki.get('createdAt')).getTime())
      / 60
      / 60
      / 1000
      / 24;
    createdElapsedDays = parseInt(createdElapsedDaysUnparsed, 10) + 1;

    return (
      <Grid.Row>
        <Card
          fluid
          // id="modal-anki"
          data-role="modal-anki"
          style={{
            minHeight: 300
          }}
        >
          <Container>
            <Grid>
              <Grid.Column floated="left" width={5} />
              <Grid.Column floated="right" width={5} textAlign="right">
                {currentAnki.size > 0 && (
                  <React.Fragment>
                    <span className="pull-right bg-green label">
                      {revisionAnkisTotal}
                    </span>

                    <span className="pull-right bg-green label">
                      {'Created: '}
                      {createdElapsedDays}
                      {'d Ago'}
                    </span>

                    <span className="pull-right bg-green label">
                      {`Rev: ${revisionElapsedDays} - ${currentAnki.getIn([
                        'revision',
                        'round'
                      ])}/${currentAnki.getIn(['revision', 'passing'])}`}
                    </span>

                    {currentAnki.has('tags') && currentAnki.get('tags').map((tag, index) => (
                      <span
                        className="pull-right bg-orange label"
                        key={tag.get('_id')}
                      >
                        {tag.get('name')}
                      </span>
                    ))}
                  </React.Fragment>
                )}
              </Grid.Column>
            </Grid>
          </Container>

          <Header as="h3" textAlign="center">
            {currentAnki.get('question')}
          </Header>

          {showAnswer
            && (
              <Container text textAlign="center">
                {currentAnki.get('answer')}
              </Container>
            )
          }

          <Divider />

          <Container textAlign="center">
            {!showAnswer && (
              <Button
                primary
                onClick={this.toggleAnswer}
              >
                SHOW ANSWER
              </Button>
            )}

            {showAnswer && (
              <React.Fragment>
                <Button
                  className="btn-large"
                  onClick={event => this.updateAnki(event, 1)}
                >
                  PASS
                </Button>

                <Button
                  color="green"
                  onClick={evnet => this.updateAnki(event, 2)}
                >
                  GOOD
                </Button>
              </React.Fragment>
            )}
          </Container>
        </Card>
      </Grid.Row>
    );
  }
}

AnkiPage.defaultProps = {
  currentAnki: Map(),
  revisionAnkisTotal: 0
};

AnkiPage.propTypes = {
  currentAnki: PropTypes.object,
  revisionAnkisTotal: PropTypes.number,

  AnkiActions: PropTypes.object.isRequired
};

function mapStateToProps(state, ownProps) {
  return {
    currentAnki: currentAnkiSelector(state),
    revisionAnkisTotal: revisionAnkisTotalSelector(state)
  };
}

function mapDispatchToProps(dispatch) {
  return {
    AnkiActions: bindActionCreators(AnkiActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AnkiPage);
