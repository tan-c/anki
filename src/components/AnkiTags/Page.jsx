import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map, List } from 'immutable';

import { AnkiActions } from 'utility-redux/anki';
import { AnkiTagActions } from 'utility-redux/ankiTag';
// import { UiActions } from 'utility-redux/ui';
import ReactTable from 'react-table';

import {
  Form, TextArea, Grid, Segment,
  Menu,
  Icon,
  Header
} from 'semantic-ui-react';

export class AnkiTagsPage extends React.Component {
  // constructor(props, context) {
  //   super(props, context);
  //   this.state = {
  //     notebookGroupIdFilter: '',
  //     tagFilter: '',
  //   };
  // }

  state = {
    selectedAnkiTagId: '',
  }

  componentWillReceiveProps = (nextProps) => {
    // if (nextProps.ankis.length) {
    //   const { notebookGroupIdFilter, tagFilter } = this.state;
    //
    //   let filteredAnkis = nextProps.ankis;
    //   if (notebookGroupIdFilter !== '') {
    //     filteredAnkis = filteredAnkis.filter(a =>
    //       a.note.notebook.notebookGroup._id === notebookGroupIdFilter);
    //   }
    //
    //   if (tagFilter !== '') {
    //     filtered Ankis = filteredAnkis.filter((a) => {
    //       let res = false;
    //       a.tags.forEach((tag) => {
    //         if (tag._id === tagFilter) {
    //           res = true;
    //         }
    //       });
    //       return res;
    //     });
    //   }
    //   filteredAnkis.sort((a, b) => (a._id >= b._id ? 1 : -1));
    //
    //   this.setState({
    //     filteredAnkis,
    //   });
    // }
  };

  createAnkiTagKeydown = (event) => {
    if (event.which === 13) {
      this.props.AnkiTagActions.create({
        name: event.target.value
      });
    }
  };

  filterAnki = (event, field) => {
    const val = event.target.value;
    const { ankis } = this.props;
    let filteredAnkis = [];
    if (val.length) {
      filteredAnkis = ankis.filter((a) => {
        let res = false;
        if (field === 'notebookGroups') {
          // this.setState({
          //   notebookGroupIdFilter: val,
          // });
          return a.note.notebook.notebookGroup._id === val;
        }

        if (field === 'tag') {
          // this.setState({
          //   tagFilter: val,
          // });
          a.tags.forEach((tag) => {
            if (tag._id === val) {
              res = true;
            }
          });
        }

        return res;
      });
    } else {
      filteredAnkis = ankis;
    }
  };

  addTagToAnki = (anki) => {
    const { selectedAnkiTagId } = this.state;

    if (
      anki.has('tags')
      && anki.get('tags').find(a => a.get('_id') === selectedAnkiTagId)
      === undefined
    ) {
      this.props.AnkiActions.update(
        anki.update('tags', tags => tags.push(selectedAnkiTagId))
      );
    }
  };

  render() {
    const {
      ankis, ankiTags, notebookGroups
    } = this.props;

    const { selectedAnkiTagId } = this.state;

    return (
      <Grid.Row>
        <Grid.Column width={2}>
          <Header as="h3" inverted>
            Tags
          </Header>

          <Menu.Menu>
            {ankiTags.valueSeq().map(tag => (
              <Menu.Item
                key={tag.get('_id')}
              >
                {tag.get('name')}
                <Icon
                  name="close"
                  onClick={(_) => {
                    this.props.AnkiTagActions.deleteRecord(tag);
                  }}
                />
              </Menu.Item>
            ))}
          </Menu.Menu>
          <input
            type="text"
            className="border-green margin-top-10"
            onKeyDown={this.createAnkiTagKeydown}
            placeholder="new anki"
          />
        </Grid.Column>

        <Grid.Column width={12}>
          <select
            onChange={event => this.filterAnki(event, 'notebookGroups')}
            className="form-control"
          >
            <option key="" value="">
              Select Notebook
            </option>
            {notebookGroups.valueSeq().map(nbg => (
              <option key={nbg.get('_id')} value={nbg.get('_id')}>
                {nbg.get('title')}
              </option>
            ))}
          </select>

          <ReactTable
            data={ankis.valueSeq().toJS()}
            style={{
              height: 'calc(100vh - 150px)',
              overflow: 'auto',
              background: 'white',
              color: 'black'
            }}
            columns={[
              {
                Header: 'Type',
                accessor: 'type',
                width: 80
              },
              {
                Header: 'Created',
                accessor: 'createdAt',
                width: 150
              },
              {
                Header: 'Question',
                accessor: 'question',
                width: 120
              }, {
                Header: 'Tags',
                accessor: 'tags',
                width: 80,
                // sortMethod: numericSortSmallerFirst,
                Cell: row => (
                  <div>
                    {row.tags && row.tags.map(tag => (
                      <span
                        key={tag.get('_id')}
                        className="bg-green color-white margin-right-5 padding-vertical-5"
                      >
                        {tag.name}
                      </span>
                    ))}
                  </div>
                )
              },
              {
                Header: 'Add Tag',
                accessor: 'add Tags',
                width: 80,
                // sortMethod: numericSortSmallerFirst,
                Cell: row => (
                  <div>
                    {selectedAnkiTagId.length > 0 && (
                      <button
                        className="bg-green margin-right-10"
                      // FIXME: actions not right
                      // onClick={(_) => {
                      //   this.addTagToAnki(row);
                      // }}
                      >
                        Add Tag
                      </button>
                    )}

                    <button
                      className="bg-red"
                    // FIXME: actions not right
                    // onClick={(_) => {
                    //   this.props.AnkiActions.update(row.set('tags', List()));
                    // }}
                    >
                      Clear Tag
                    </button>
                  </div>
                ),
              } // {
              //   Header: 'Answer',
              //   accessor: 'answer',
              //   width: 120
              // }
            ]}

            // defaultPageSize={30}
            filterable
            defaultSorted={[
              // {
              //   id: 'itemType',
              //   desc: true
              // },
              // {
              //   id: 'price',
              //   desc: true
              // }
            ]}
          />

          {/* {ankis.valueSeq().map(anki => (
            <div
              key={anki.get('_id')}
              className="flex-container-row border-bottom-black"
            >
              <span className="flex-1 border-right-black-20">

              </span>
              <span className="flex-1">

              </span>
            </div>
          ))} */}
        </Grid.Column>
      </Grid.Row>
    );
  }
}

AnkiTagsPage.defaultProps = {
  ankis: Map(),
  ankiTags: Map()
};

AnkiTagsPage.propTypes = {
  ankis: PropTypes.object,
  ankiTags: PropTypes.object,
  notebookGroups: PropTypes.object.isRequired,
  // selectedAnkiTagId: PropTypes.string.isRequired,

  AnkiActions: PropTypes.object.isRequired,
  AnkiTagActions: PropTypes.object.isRequired
};

function mapStateToProps(state, ownProps) {
  return {
    ankis: state.ankis,
    ankiTags: state.ankiTags,
    notebookGroups: state.notebookGroups,
    // selectedAnkiTagId: state.ui.getIn(['selectedAnkiTagId'])
  };
}

function mapDispatchToProps(dispatch) {
  return {
    AnkiActions: bindActionCreators(AnkiActions, dispatch),
    AnkiTagActions: bindActionCreators(AnkiTagActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AnkiTagsPage);
