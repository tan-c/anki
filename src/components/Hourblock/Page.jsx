import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import moment from 'moment-timezone';
// import { Map } from 'immutable';
import { currentUserSelector } from 'utility-redux/user';
import { totalProjectTasksCountSelector } from 'utility-redux/task';
import DailyRecordsChartConnected from './DailyRecordsChart';
// import { UiActions } from 'utility-redux/ui';

import HourBlockListConnected from './List';

import EventDetailsConnected from './ExtraSection/Event/Details';

import ProjectTaskListConnected from './ExtraSection/Task/ProjectTaskList';
import FocusedProjectTaskListConnected from './ExtraSection/Task/FocusedProjectTaskList';

import DailyWorkoutRecordsDetailsConnected from './ExtraSection/Daily/WorkoutRecordsDetails';
import DailyCalorieDetailsConnected from './ExtraSection/Daily/CalorieDetails';
import DailySleepDetailsConnected from './ExtraSection/Daily/SleepDetails';
import DailyMeasurementsConnected from './ExtraSection/Daily/Measurements';
import CategoryInsightsConnected from './LeftAside/CategoryInsights';
import WeeklyInsightsConnected from './LeftAside/WeeklyInsights';

import RightAsideConnected from './RightAside';
import FooterConnected from './Footer';

export class HourBlockPage extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      showingItemType: 'daily',
      dayObjectOffset: 0,
      // NOTE - This should be based on localtimezone
      dayMomentObject: moment().startOf('day'),
    };
  }

  updateDayMomentObject = (offset) => {
    const { dayObjectOffset } = this.state;
    const newDayObjectOffset = offset === 0 ? offset : dayObjectOffset + offset;
    this.setState({
      dayObjectOffset: newDayObjectOffset,
      // NOTE - This should be based on localtimezone
      dayMomentObject: moment().startOf('day').add(newDayObjectOffset, 'day'),
    });
  }

  // {loadingPomoList &&
  //   <div className="loading-spinner">
  //     <i className="fa fa-cog fa-spin fa-1x fa-fw" />
  //   </div>
  // }

  renderHourBlockList = () => {
    const { dayMomentObject } = this.state;
    return (
      <Fragment>
        {!window.isMobile && (
          <div className="bg-white-20 height-40 flex-container-row">
            <span
              className={`flex-1 text-left height-40 line-height-40 font-400 font-24 margin-left-10 ${dayMomentObject.dayOfYear() === moment().dayOfYear() ? 'color-green' : ''}`}
            >
              {dayMomentObject.format('YYYY-MM-DD ddd')}
              {' '}
              - W
              {dayMomentObject.isoWeek()}
            </span>

            <div className="flex-1 text-right margin-right-10">
              <div className="font-400 color-green height-20 line-height-20">QUICK LINK</div>
              <br />
              <div className="height-20 line-height-20">
                <span role="button" tabIndex="-1" onClick={_ => this.updateDayMomentObject(-1)}>Last Day</span>
                <span>|</span>
                <span role="button" tabIndex="-1" onClick={_ => this.updateDayMomentObject(0)}>Today</span>
                <span>|</span>
                <span role="button" tabIndex="-1" onClick={_ => this.updateDayMomentObject(1)}>Next Day</span>
              </div>
            </div>
          </div>
        )}

        <HourBlockListConnected dayMomentObject={dayMomentObject} />
      </Fragment>
    );
  }

  renderRightSection = () => {
    const { dayMomentObject, showingItemType } = this.state;
    const { totalProjectTasksCount } = this.props;

    return (
      <Fragment>
        <span className="flex-container-row typical-setup border-green">
          <span
            className={`flex-1 ${showingItemType === 'tasks' ? 'bg-green' : ''} ${showingItemType === 'tasks' && totalProjectTasksCount > 0 ? 'bg-red' : ''}`}
            role="button"
            tabIndex="-1"
            onClick={_ => this.setState({ showingItemType: 'tasks' })}
          >
            {`Task - ${totalProjectTasksCount}`}
          </span>
          <span
            className={`border-right-green flex-1 ${showingItemType === 'daily' && 'bg-green'}`} role="button" tabIndex="-1"
            onClick={_ => this.setState({ showingItemType: 'daily' })}
          >
            Daily
          </span>
          <span
            className={`border-right-green flex-1 ${showingItemType === 'gym' && 'bg-green'}`} role="button" tabIndex="-1"
            onClick={_ => this.setState({ showingItemType: 'gym' })}
          >
            Gym
          </span>
        </span>

        {showingItemType === 'tasks'
          && (
            <React.Fragment>
              <section className="flex-1">
                <ProjectTaskListConnected />
              </section>
              <section className="flex-1">
                <FocusedProjectTaskListConnected />
              </section>
            </React.Fragment>
          )
        }

        {showingItemType === 'daily'
          && (
            <React.Fragment>
              <section className="flex-2">
                <DailyMeasurementsConnected dayMomentObject={dayMomentObject} showingItemType={showingItemType} />
                <DailySleepDetailsConnected dayMomentObject={dayMomentObject} />
                <div className="spacing-10" />
                <DailyCalorieDetailsConnected dayMomentObject={dayMomentObject} />
              </section>
              {!window.isMobile && <DailyRecordsChartConnected />}
            </React.Fragment>
          )

        }

        {showingItemType === 'gym'
          && (
            <section className="flex-2">
              <DailyMeasurementsConnected dayMomentObject={dayMomentObject} showingItemType={showingItemType} />
              <div className="spacing-10" />
              <DailyWorkoutRecordsDetailsConnected dayMomentObject={dayMomentObject} />
            </section>
          )
        }

        {!window.isMobile && <EventDetailsConnected />}
      </Fragment>
    );
  }

  render() {
    const { showTaskSection } = this.props;

    if (window.isMobile) {
      return (
        <div className="flex-container page" data-role="hourblock-page">
          {showTaskSection && (
            <section className="flex-container flex-1 no-bg">
              {this.renderRightSection()}
            </section>
          )}

          <section className="flex-1">
            {this.renderHourBlockList()}
          </section>
        </div>
      );
    }

    return (
      <main id="hourblock">
        <aside id="left-menu" className={`${window.isMobile && 'mobile'}`}>
          <CategoryInsightsConnected />
          <WeeklyInsightsConnected />
        </aside>

        <RightAsideConnected />
        <FooterConnected />

        <div
          className="flex-container-row page" data-role="hourblock-page"
        >
          <section className="flex-5">
            {this.renderHourBlockList()}
          </section>

          {showTaskSection && (
            <section className="flex-container flex-2 no-bg">
              {this.renderRightSection()}
            </section>
          )}
        </div>
      </main>
    );
  }
}

HourBlockPage.defaultProps = {
  showTaskSection: false,
  totalProjectTasksCount: 0
};

HourBlockPage.propTypes = {
  showTaskSection: PropTypes.bool,
  totalProjectTasksCount: PropTypes.number,
};

function mapStateToProps(state, ownProps) {
  return {
    showTaskSection: currentUserSelector(state).hasIn(['config', 'showTaskSection']) && currentUserSelector(state).getIn(['config', 'showTaskSection']),
    totalProjectTasksCount: totalProjectTasksCountSelector(state),
  };
}

export default connect(mapStateToProps)(HourBlockPage);
