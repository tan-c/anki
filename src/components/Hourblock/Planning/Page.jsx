import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';
import moment from 'moment-timezone';

import { UiActions } from 'utility-redux/common/ui';
import { PlannedPomoActions, plannedPomoByDayOfWeekSelector } from 'utility-redux/hourblock/plannedPomo';
import PlanningItemConnected from './Item';

export class PlanningPage extends React.Component {
  // constructor(props, context) {
  //   super(props, context);
  // }

  getHourblocks = () => {
    const hourblocks = [];
    for (let i = 0; i < 48; i += 1) {
      hourblocks.push(i);
    }
    return hourblocks;
  }

  updatePlannedPomo = (hourblock, day, isDeleting = false) => {
    const { plannedPomos, selectedProjectId } = this.props;
    const dayPlannedPomo = plannedPomos.get(day.toString());
    let newDayPlannedPomo = null;

    if (isDeleting) {
      this.props.UiActions.updateIn(['hourblock', 'planningPage', 'updatingPlannedPomo'], true);
      newDayPlannedPomo = dayPlannedPomo.setIn(['plannedPomos', hourblock.toString(), 'project'], null);
    } else {
      if (!selectedProjectId.length) {
        return;
      }
      this.props.UiActions.updateIn(['hourblock', 'planningPage', 'updatingPlannedPomo'], true);
      const originalPomo = dayPlannedPomo.getIn(['plannedPomos', hourblock.toString()]);
      if (originalPomo === null) {
        newDayPlannedPomo = dayPlannedPomo.setIn(['plannedPomos', hourblock.toString()], Map({
          project: selectedProjectId,
        }));
      } else {
        newDayPlannedPomo = dayPlannedPomo.setIn(['plannedPomos', hourblock.toString(), 'project'], selectedProjectId);
      }
    }

    this.props.PlannedPomoActions.update(newDayPlannedPomo).then((res) => {
      this.props.UiActions.updateIn(['hourblock', 'planningPage', 'updatingPlannedPomo'], false);
    });
  }

  updateRecurTask = (event, day, hourblock) => {
    if (event.keyCode === 13) {
      const { plannedPomos } = this.props;
      const plannedPomoDay = plannedPomos.get(day.toString());
      const newItem = plannedPomoDay.setIn(['plannedPomos', hourblock.toString(), 'tasks', 'recur'], event.target.value);
      this.props.PlannedPomoActions.update(newItem);
    }
  }

  render() {
    const { plannedPomos } = this.props;

    return (
      <div data-role="planning-page" className="page">
        <section className="flex-1">
          <div className="section-header">
            <span className="flex-container-row border-bottom">
              <span className="flex-1 border-right" />
              {plannedPomos.keySeq().map((key, index) => (
                <span className="flex-4 border-right" key={key}>
                  {parseInt(key, 10) + 1}
                </span>))}
            </span>
          </div>

          <div className="section-content">
            {this.getHourblocks().map(hourblock => (
              <span className={`flex-container-row typical-setup border-bottom ${hourblock % 2 === 0 && 'border-top'} ${hourblock % 4 === 0 && 'border-top-white'}`} key={hourblock}>
                <span className="width-50">
                  {moment().tz('Asia/Tokyo').startOf('day').add(hourblock / 2, 'hour')
                    .format('HH:mm')}
                </span>

                {[0, 1, 2, 3, 4, 5, 6].map(day => (
                  <PlanningItemConnected
                    key={day}
                    hourblock={hourblock}
                    day={day}
                    plannedPomo={plannedPomos.getIn([day.toString(), 'plannedPomos', hourblock.toString()]) || Map()}
                    updatePlannedPomo={this.updatePlannedPomo}
                    updateRecurTask={this.updateRecurTask}
                  />
                ))}
              </span>))}
          </div>
        </section>
      </div>
    );
  }
}

PlanningPage.defaultProps = {
  selectedProjectId: '',
};

PlanningPage.propTypes = {
  plannedPomos: PropTypes.object.isRequired,
  selectedProjectId: PropTypes.string,

  UiActions: PropTypes.object.isRequired,
  PlannedPomoActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    plannedPomos: plannedPomoByDayOfWeekSelector(state),
    selectedProjectId: state.ui.getIn(['hourblock', 'planningPage', 'selectedProjectId']),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    UiActions: bindActionCreators(UiActions, dispatch),
    PlannedPomoActions: bindActionCreators(PlannedPomoActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(PlanningPage);
