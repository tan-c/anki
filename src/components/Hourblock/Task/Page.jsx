import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';
import moment from 'moment-timezone';

import Input from 'utility-react-component/Form/Input/Uncontrolled';
import InputNewConnected from 'utility-react-component/Form/Input/New';
import ProjectSelectConnected from 'utility-react-component/Form/HourblockProjectSelect';

import { TaskActions, currentYearlyTasksSortedSelector } from 'utility-redux/hourblock/task';
import { UiActions } from 'utility-redux/common/ui';

import MonthlyTasksListConnected from './MonthlyTasks/List';
import DailyTaskPlanningConnected from './DailyTasks/List';

export class TasksPage extends React.Component {
  // constructor(props, context) {
  //   super(props, context);
  // }
  //
  // componentWillReceiveProps(nextProps) {
  //   this.setState({
  //   });
  // }

  render() {
    const { currentYearlyTasksSorted } = this.props;

    return (
      <div data-role="task-page" className="flex-container-row page">
        <section className="flex-1">
          <div className="section-header">2018 Tasks</div>
          <div className="section-content">
            <div className="list-with-pinned-bottom">
              {currentYearlyTasksSorted.size > 0 && currentYearlyTasksSorted.map(task => (
                <div key={task.get('_id')} className="flex-container-row typical-setup">
                  <span className="flex-1">
                    <ProjectSelectConnected
                      onChangeEvent={event => this.props.TaskActions.update(task.set('project', event.target.value), task)}
                      value={task.getIn(['project', '_id'])}
                      color={task.getIn(['project', 'category', 'color'])}
                    />
                  </span>
                  <Input
                    inputName="content"
                    inputClassNames="flex-2"
                    record={task}
                    actions={this.props.TaskActions}
                  />
                  <i
                    className="fa fa-fw fa-eye"
                    role="button"
                    tabIndex="-1"
                    onClick={_ => this.props.UiActions.updateIn(['hourblock', 'taskPage', 'selectedYearlyTaskId'], task.get('_id'))}
                  />
                </div>))}
            </div>

            <div className="flex-container-row pinned-bottom border-top">
              <InputNewConnected
                inputName="content"
                newRecord={{
                  targetCompletion: moment().tz('Asia/Tokyo').startOf('year'),
                  type: 'yearly',
                }}
                actions={this.props.TaskActions}
              />
            </div>
          </div>
        </section>

        <MonthlyTasksListConnected />
        <DailyTaskPlanningConnected />
      </div>
    );
  }
}

TasksPage.defaultProps = {
  currentYearlyTasksSorted: Map(),
};

TasksPage.propTypes = {
  currentYearlyTasksSorted: PropTypes.object,

  TaskActions: PropTypes.object.isRequired,
  UiActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    currentYearlyTasksSorted: currentYearlyTasksSortedSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    TaskActions: bindActionCreators(TaskActions, dispatch),
    UiActions: bindActionCreators(UiActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(TasksPage);
