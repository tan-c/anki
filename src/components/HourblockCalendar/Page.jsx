import React from 'react';
// import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
// import { Map } from 'immutable';
import moment from 'moment-timezone';

import { Grid } from 'semantic-ui-react';
import MonthlyTasksListConnected from './MonthlyTasksList';
import DailyTasksListConnected from './DailyTasksList';


export class RightAside extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      activeMonth: moment().tz('Asia/Tokyo').month(),
      todayMoment: moment().tz('Asia/Tokyo').startOf('day'),
    };
  }

  // componentWillReceiveProps(nextProps) {
  //   this.setState({
  //   });
  // }

  // filter(dayVal => !dayVal.isPast || dailyTasks.has(dayVal.dayOfYearString)

  createMonthlyList = () => {
    const { todayMoment } = this.state;
    const monthList = [];
    for (let i = 0; i < 12; i += 1) {
      const currentMonth = [];
      for (let j = 0; j < moment().tz('Asia/Tokyo').startOf('year').add(i, 'month')
        .daysInMonth(); j += 1) {
        const dayMomentObject = moment().tz('Asia/Tokyo').startOf('year').add(i, 'month')
          .add(j, 'day');
        currentMonth.push({
          dayMomentObject,
          dayOfYearString: dayMomentObject.dayOfYear().toString(),
          weekString: dayMomentObject.isoWeek().toString(),
          isPast: dayMomentObject.valueOf() < todayMoment.valueOf(),
          isWeekend: dayMomentObject.day() === 0 || dayMomentObject.day() === 6,
        });
      }
      monthList.push(currentMonth);
    }

    return monthList;
  }

  render() {
    const { activeMonth } = this.state;
    const monthList = this.createMonthlyList();

    return (
      <Grid.Row>
        <Grid.Column>
          <span className="flex-container-row">
            {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12].map(month => (
              <div
                key={month}
                role="button" tabIndex="-1"
                className={`flex-1 ${activeMonth === month - 1 && 'color-orange font-400'}`}
                onClick={_ => this.setState({
                  activeMonth: month - 1,
                })}
              >
                {month}
              </div>))}
          </span>

          <MonthlyTasksListConnected activeMonth={activeMonth} />

          {monthList[activeMonth].map(dayVal => (
            <DailyTasksListConnected
              key={dayVal.dayMomentObject.unix()}
              dayVal={dayVal}
            />
          ))}
        </Grid.Column>
      </Grid.Row>
    );
  }
}

export default RightAside;
