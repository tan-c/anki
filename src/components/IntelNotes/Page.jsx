// // import {Link} from 'react-router';
// import { Map } from 'immutable';

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import NoteEditorConnected from './NoteEditor/NoteEditor';
import NoteListConneted from './Note/List';
import NotebookGroupListConnected from './NotebookGroup/List';

export class IntelNotePage extends React.Component {
  render() {
    return (
      <React.Fragment>
        {!window.isMobile
          && (
            <React.Fragment>
              <NotebookGroupListConnected />
              <NoteListConneted />
              <NoteEditorConnected />
            </React.Fragment>
          )
        }
      </React.Fragment>
    );
  }
}

IntelNotePage.defaultProps = {
};

IntelNotePage.propTypes = {
  // ui: PropTypes.object,
};

function mapStateToProps(state, ownProps) {
  return {
  };
}


function mapDispatchToProps(dispatch) {
  return {
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(IntelNotePage);
