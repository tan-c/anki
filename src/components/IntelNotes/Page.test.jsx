import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { IntelNotePage } from './Page';

function setup() {
  const props = {
  };

  return shallow(<IntelNotePage {...props} />);
}

describe('IntelNotePage Unit Tests', () => {
  describe('', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      // expect(wrapper.find("div[data-role='anki-page']")).toHaveLength(1);
    });
  });
});
