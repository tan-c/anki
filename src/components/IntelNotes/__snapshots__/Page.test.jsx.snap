// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntelNotePage Unit Tests  renders correctly 1`] = `
<Fragment>
  <Connect(NotebookGroupList) />
  <Connect(NoteList) />
  <Connect(NoteEditor) />
</Fragment>
`;
